scripts/build/dynbinary

```diff
@@ -9,6 +9,14 @@ source ./scripts/build/.variables
 
 echo "Building dynamically linked $TARGET"
 export CGO_ENABLED=1
-go build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" -buildmode=pie "${SOURCE}"
+GO_BUILDMODE="-buildmode=pie"
+#pie build mode is not supported for the mips arch 
+for arch in mips mipsle mips64 mips64le ppc64
+do
+    if [ "$(go env GOARCH)" = "${arch}" ];then
+        GO_BUILDMODE=""
+    fi
+done
+go build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" "${GO_BUILDMODE}" "${SOURCE}"
```

Comment:
Perhaps use a `case`, so that we don't have to do this loop;

 ```suggestion

case "$(go env GOARCH)" in
    mips*|ppc64)
        # pie build mode is not supported on mips architectures
        GO_BUILDMODE=""
        ;;
    *)
        GO_BUILDMODE="-buildmode=pie"
        ;;
esac

go build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" "${GO_BUILDMODE}" "${SOURCE}"
```

Reply:
Agreed, also `${GO_BUILDMODE}` can't be quoted in this final `go build` line (or it'll pass the empty saying as an argument).

Reply:
Good catch; I was looking at it, but didn't think about the "empty" case ❤️

Reply:
(updated my example/suggestion)

Reply:
I  have learnt a lot，thanks！：）